aws/rust-runtime/aws-types/src/sdk_config.rs

```diff
@@ -558,8 +555,8 @@ impl SdkConfig {
 
     #[doc(hidden)]
     /// Configured sleep implementation
-    pub fn sleep_impl(&self) -> Option<Arc<dyn AsyncSleep>> {
-        self.sleep_impl.clone()
+    pub fn sleep_impl(&self) -> Option<&SharedAsyncSleep> {
```

Comment:
Better to just clone it since it's reference counted. Otherwise the caller has to clone it if they want to keep it.

Reply:
Yeah, I suppose it's sort of "cheating" to use a reference counted object without participating in shared ownership. Will return a clone of it.

Reply:
Addressed in c86ce4b.